accepted/0042-isolated-mode.md

```diff
@@ -0,0 +1,347 @@
+
+## Prior Art and Alternatives
+
+### [pnpm](https://github.com/pnpm/pnpm)
+
+Package manager with a `isolated-mode`.
+This RFC is mostly inspired by pnpm.
+
+### [ied](https://github.com/alexanderGugel/ied)
+
+Package manager with a `isolated-mode`.
+Similar to pnpm but unmaintained.
+
+### [nix](https://nixos.wiki/wiki/Nix)
+
+Functional package manager. Can work with npm packages (eg. [node2nix](https://github.com/svanderburg/node2nix) or [nixfromnpm](https://github.com/adnelson/nixfromnpm) )
+
+### [yarn](https://yarnpkg.com/)
+
+Package manager with a `isolated-mode` and project manager.
+
+### [Import maps](https://github.com/WICG/import-maps)
+
+Standard supported by [a few browsers](https://caniuse.com/import-maps) and [deno](https://deno.land/) which makes it possible to implement `isolated-mode`.
+
+## Unresolved Questions and Bikeshedding
+
+- Should we use symlinks or junctions on Windows? Both of them have drawbacks:
+
+  - Junctions have to be representated by an absolute path, this means that junctions cannot be committed to git or packed into a package.
+  - Symlinks can only be created in elevated shell [or when Windows is in "developer mode"](https://blogs.windows.com/windowsdeveloper/2016/12/02/symlinks-windows-10/#LCiVBWTgQF5s7fmL.97).
+
+  - answer: junctions by default. support for symlinks can be added later as an opt-in if we see value for it.
+
+- How much community code will break when the system forbids access to undeclared dependencies? In other words, how much code needs to be fixed to work properly in `isolated-mode`?
+  - Regarding packages with missing dependencies in the `package.json` file. In many cases, package owners are willing to fix such issues. If there are a significant number of issues encountered around this, we can likely expect the of packages with missing dependencies is expected to drop fast after npm release `isolated-mode`. We may want to later add a feature to npm which allows users to locally declare dependencies on behalf of packages as a stop-gap, if existing solutions to this are not enough.
+  - There don't seem to be any packages out there that depend on the way hoisting works as a feature. If this is the case, it would be easy to argue that these packages should only rely on the npm and the Node.js contracts/APIs and not on implementation details like hoisting.
+  - Some dev-environments don't support symlinks.
+    - AWS Lambdas -> a repository can a be installed with isolated mode locally and on CI but then deployed in hoisted mode.
+    - React native -> There are plugins existing to make react-native work with symlinks
+  - If a package is missing a dependency, it can be temporarily fixed (while waiting for the package owner to fix this issue) by declaring this missing dependency as top level dependency of the repository.
```

Comment:
```suggestion
  - If a package is missing a dependency, it can be temporarily fixed (while waiting for the package owner to fix this issue, or if it’s working as designed) by declaring this missing dependency as top level dependency of the repository.
```